Yosemite/Yosemite/Model/Payments/CardPresentPaymentsConfiguration.swift

```diff
@@ -1,64 +1,83 @@
+        self.countryCode = countryCode
+        self.stripeTerminalforCanadaEnabled = stripeTerminalforCanadaEnabled
         self.paymentMethods = paymentMethods
         self.currencies = currencies
         self.paymentGateways = paymentGateways
         self.supportedReaders = supportedReaders
-        self.purchaseCardReaderUrl = purchaseCardReaderUrl
     }
 
     public init(country: String, canadaEnabled: Bool) {
         switch country {
         case "US":
-            //TODO: update to use Self.purchaseCardReaderUrl(for: country) when pages/redirects are added to the website pdfdoF-su-p2
             self.init(
+                countryCode: country,
+                stripeTerminalforCanadaEnabled: canadaEnabled,
                 paymentMethods: [.cardPresent],
                 currencies: ["USD"],
                 paymentGateways: [WCPayAccount.gatewayID, StripeAccount.gatewayID],
-                supportedReaders: [.chipper, .stripeM2],
-                purchaseCardReaderUrl: Constants.purchaseM2ReaderUrl
+                supportedReaders: [.chipper, .stripeM2]
             )
         case "CA" where canadaEnabled == true:
             self.init(
+                countryCode: country,
+                stripeTerminalforCanadaEnabled: true,
                 paymentMethods: [.cardPresent, .interacPresent],
                 currencies: ["CAD"],
                 paymentGateways: [WCPayAccount.gatewayID],
-                supportedReaders: [.wisepad3],
-                purchaseCardReaderUrl: Self.purchaseCardReaderUrl(for: country)
+                supportedReaders: [.wisepad3]
             )
         default:
-            self.init(paymentMethods: [],
-                      currencies: [],
-                      paymentGateways: [],
-                      supportedReaders: [],
-                      purchaseCardReaderUrl: Constants.fallbackInPersonPaymentsUrl)
+            self.init(
+                countryCode: country,
+                stripeTerminalforCanadaEnabled: canadaEnabled,
+                paymentMethods: [],
+                currencies: [],
+                paymentGateways: [],
+                supportedReaders: []
+            )
         }
     }
 
     public var isSupportedCountry: Bool {
         paymentMethods.isEmpty == false && currencies.isEmpty == false && paymentGateways.isEmpty == false && supportedReaders.isEmpty == false
     }
 
-    private static func purchaseCardReaderUrl(for countryCode: String) -> URL {
-        URL(string: Constants.purchaseReaderForCountryUrlBase + countryCode) ?? Constants.fallbackInPersonPaymentsUrl
+    /// Given a two character country code and the active plugin, returns a URL
+    /// where the merchant can purchase a card reader
+    ///
+    public func purchaseCardReaderUrl(for plugin: CardPresentPaymentsPlugins) -> URL {
+        if case .stripe = plugin {
+            return Constants.stripeReaderPurchaseUrl
+        }
+
+        if !stripeTerminalforCanadaEnabled {
```

Comment:
I know this is temporary but we probably should still use the M2 URL for US stores, even if the Canada flag is enable. For reference, the country-based US URL redirects to this page 😬:

![Simulator Screen Shot - iPhone 13 - 2022-02-28 at 11 20 52](https://user-images.githubusercontent.com/8739/155966164-88ff4d8b-7129-414d-832d-df5838a36cd0.png)